### ISSUE TITLE
filling sequence error

### ISSUE THREAD

**Issue by xinkunlong** [TexZK/hexrec#14]
```python
import hexrec.records as hr

default_values = [0xFF for _ in range(256)]

hr.save_chunk('TEST.s19', bytes(default_values), 0X1f00)

memory = hr.load_memory('TEST.s19')

memory.fill(0X1f00,0X1f05,bytes([0x01,0x02,0x03,0x04,0x05]))

hr.save_memory('TEST.s19', memory)
````
the first position is 0x02( position: 0x1f00 )
the last position is 0x01(position: 0x1f04)

the sequence is 0x02 0x03 0x04 0x05 0x01

**Comment by TexZK** [comment]
Hello. This is the expected behavior as per the current implementation.
The ``pattern`` argument of ``fill()`` and ``flood()`` methods is currently **repeated/wrapped with respect to address zero**, instead of the ``start`` argument.

I agree that this behavior is rather counterintuitive, and **I should change this**; for sure, I should document this better.
After the adaptation to the ``start`` argument, the only undefined case is when ``start`` is ``None``. In that case, its value should be replaced by the actual memory start address (computed by the ``memory.start`` property).

Anyway, as a workaround for your specific case, you can use the ``write()`` method: there is no need to use ``fill()`` with a ``pattern`` exactly the same size of the area to be written.

```py
memory.write(0x1F00, bytes([0x01, 0x02, 0x03, 0x04, 0x05]))
```

**Comment by TexZK** [comment]
Proposal implemented in commit 22c68495b26dc13ca00aecae2faf70dc8062c662

**Comment by TexZK** [comment]
Please update your ``hexrec`` package to the latest release: _v0.2.3_.
